### PR TITLE
Implement window command timeouts

### PR DESCRIPTION
**Pull request by openbots-su**

### CODE REVIEW

File: OpenBots.Commands/OpenBots.Commands.UIAutomation/OpenBots.Commands.Window/ActivateWindowCommand.cs [hunk 1]
```diff
@@ -28,27 +29,53 @@ public class ActivateWindowCommand : ScriptCommand
 		[Editor("CaptureWindowHelper", typeof(UIAdditionalHelperType))]
 		public string v_WindowName { get; set; }
 
+		[Required]
+		[DisplayName("Timeout")]
+		[Description("Set a timeout in seconds.")]
+		[SampleUsage("30 || {vTimeout}")]
+		[Remarks("")]
+		public string v_Timeout { get; set; }
+
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.UIAutomation/OpenBots.Commands.Window/ActivateWindowCommand.cs:38]: UIAutomationCommand and various other commands have a timeout field like this one. Please use it as a template for the v_Timeout attributes

File: OpenBots.Commands/OpenBots.Commands.UIAutomation/OpenBots.Commands.Window/MoveWindowCommand.cs [hunk 2]
```diff
@@ -57,12 +65,34 @@ public MoveWindowCommand()
...
 			var engine = (IAutomationEngineInstance)sender;
 			string windowName = v_WindowName.ConvertUserVariableToString(engine);
+			int timeout = Int32.Parse(v_Timeout);
+			DateTime timeToEnd = DateTime.Now.AddSeconds(timeout);
+			List<IntPtr> targetWindows = User32Functions.FindTargetWindows(windowName);
+			while (timeToEnd >= DateTime.Now)
+            {
+				try
+				{
+					targetWindows = User32Functions.FindTargetWindows(windowName);
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.UIAutomation/OpenBots.Commands.Window/MoveWindowCommand.cs:82]: Does FindTargetWindows find a window even if it is minimized? If it's minimized then it's technically not active.

File: OpenBots.Commands/OpenBots.Commands.UIAutomation/OpenBots.Commands.Window/MoveWindowCommand.cs [hunk 3]
```diff
@@ -57,12 +65,34 @@ public MoveWindowCommand()
...
+			v_Timeout = "30";
 		}
 
 		public override void RunCommand(object sender)
 		{
 			var engine = (IAutomationEngineInstance)sender;
 			string windowName = v_WindowName.ConvertUserVariableToString(engine);
+			int timeout = Int32.Parse(v_Timeout);
+			DateTime timeToEnd = DateTime.Now.AddSeconds(timeout);
+			List<IntPtr> targetWindows = User32Functions.FindTargetWindows(windowName);
```
Comment (openbots-ff) [inline OpenBots.Commands/OpenBots.Commands.UIAutomation/OpenBots.Commands.Window/MoveWindowCommand.cs:77]: should this be outside of the while loop as well?

**Review by openbots-ff — CHANGES_REQUESTED** [review]

**Review by openbots-ch — DISMISSED** [review]

File: OpenBots.Commands/OpenBots.Commands.UIAutomation/OpenBots.Commands.Window/MoveWindowCommand.cs (hunk 2 above)
Comment (openbots-su) [reply OpenBots.Commands/OpenBots.Commands.UIAutomation/OpenBots.Commands.Window/MoveWindowCommand.cs:82]: From my testing, it is able to find minimized windows. The window is made active and brought to the foreground upon being moved / resized.

File: OpenBots.Commands/OpenBots.Commands.UIAutomation/OpenBots.Commands.Window/MoveWindowCommand.cs (hunk 3 above)
Comment (openbots-su) [reply OpenBots.Commands/OpenBots.Commands.UIAutomation/OpenBots.Commands.Window/MoveWindowCommand.cs]: Addressed in fe53ce7

File: OpenBots.Commands/OpenBots.Commands.UIAutomation/OpenBots.Commands.Window/ActivateWindowCommand.cs (hunk 1 above)
Comment (openbots-su) [reply OpenBots.Commands/OpenBots.Commands.UIAutomation/OpenBots.Commands.Window/ActivateWindowCommand.cs:39]: Addressed in fe53ce7

**Review by openbots-ff — APPROVED** [review]

**Review by openbots-ch — APPROVED** [review]

**Pull request merged**